src/SDL_internal.h

```diff
@@ -185,6 +185,46 @@
 
 #include <SDL3/SDL.h>
 #include <SDL3/SDL_intrin.h>
+
+
+#ifdef __ARM_NEON
+#define HAVE_NEON_INTRINSICS 1
+#endif
+
+#if defined(__MMX__) && !defined(SDL_DISABLE_MMINTRIN_H)
+#define HAVE_MMX_INTRINSICS 1
+#endif
+
+#if defined(__SSE__) && !defined(SDL_DISABLE_XMMINTRIN_H)
```

Comment:
We probably shouldn't tie XMMINTRIN_H to SSE. Is there a reason this isn't SDL_DISABLE_SSE, and then failing `#include <xmmintrin.h>` could set that?

Reply:
I honestly didn't put many thought into this and moved the code from `SDL_audiocvt.c` and `SDL_audiotypecvt.c` up to `SDL_internal.h`.
These macro (more or less) match the defines in `include/SDL3/SDL_intrin.h`:
https://github.com/libsdl-org/SDL/blob/2897de9efc926b34d3fd96a58f9a94580ffdb421/include/SDL3/SDL_intrin.h#L104-L119
On a side node, does `HAVE_IMMINTRIN_H` belong here?

Cosmopolitan libc provide these intrinsic headers, but perhaps they are incomplete. Perhaps worth an upstream bug report.

What I'm trying to do here is make the CMake system "master of its domain".
About your `SDL_DISABLE_SSE` suggestion, wouldn't introduction of another macro complicate things?
e.g. add this to `SDL_internal.h`:
```c
#if defined(__SSE__) && !defined(SDL_DISABLE_XMMINTRIN_H) && !defined(SDL_DISABLE_SSE)
#define HAVE_SSE_INTRINSICS 1
#endif
```

Reply:
No, I was thinking that SDL_DISABLE_XMMINTRIN_H would go away.

Reply:
It's gone.
The macro's are now:
- `SDL_DISABLE_NEON` for disabling arm neon
- `SDL_DISABLE_LSX` for disabling loonarch lsx
- `SDL_DISABLE_LASX` for disabling loonarch lasx
- `SDL_DISABLE_AVX` for disabling intel avx
- `SDL_DISABLE_MMX` for disabling intel mmx
- `SDL_DISABLE_SSE` for disabling intel sse
- `SDL_DISABLE_SSE2` for disabling intel sse2
- `SDL_DISABLE_SSE3` for disabling intel sse3
- `SDL_ENABLE_ALTIVEC` for **enabling** powerpc altivec


This `altivec.h` header does not cooperate well with SDL's precompiled headers.
pch is thus disabled for all sources making use of altivec (=only `src/video/SDL_blit_N.c`), and only for those sources `SDL_ENABLE_ALTIVEC`